src/providers/base.py

```
@@ -150,7 +179,45 @@ async def execute_with_retry(
+                            try:
+                                retry_after = float(header_val)
+                            except (ValueError, TypeError):
+                                pass
+
+                    if hasattr(e, 'response') and callable(getattr(e.response, 'json', None)):
+                        try:
+                            body = e.response.json()
+                            if isinstance(body, dict):
+                                limit_type = body.get('error', {}).get('type', limit_type)
+                        except Exception:
+                            pass
```

Ruff (0.14.10)

`try`-`except`-`pass` detected, consider logging the exception

(S110)